protocol/x/perpetuals/types/liquidity_tier.go

```diff
@@ -88,93 +99,43 @@ func (liquidityTier LiquidityTier) GetMaxAbsFundingClampPpm(clampFactorPpm uint3
+		return lib.BigU(liquidityTier.InitialMarginPpm)
 	}
 
-	// If `open_interest_lower_cap` < `open_interest` <= `open_interest_upper_cap`, calculate the scaled OIMF.
-	// `Scaling Factor = (Open Notional - Lower Cap) / (Upper Cap - Lower Cap)`
-	ratScalingFactor := new(big.Rat).SetFrac(
-		new(big.Int).Sub(
-			openInterestQuoteQuantums, // reuse pointer for memory efficiency
-			bigOpenInterestLowerCap,
-		),
-		bigOpenInterestUpperCap.Sub(
-			bigOpenInterestUpperCap, // reuse pointer for memory efficiency
-			bigOpenInterestLowerCap,
-		),
-	)
-
-	// `IMF Increase = Scaling Factor * (1 - Base IMF)`
-	ratIMFIncrease := lib.BigRatMulPpm(
-		ratScalingFactor,
-		lib.OneMillion-liquidityTier.InitialMarginPpm, // >= 0, since we check in `liquidityTier.Validate()`
-	)
-
-	// Calculate `Max(IMF Increase, 0)`.
-	if ratIMFIncrease.Sign() < 0 {
-		panic(
-			fmt.Sprintf(
-				"GetInitialMarginQuoteQuantums: IMF Increase is negative (%s), liquidityTier: %+v, openInterestQuoteQuantums: %s",
-				ratIMFIncrease.String(),
-				liquidityTier,
-				openInterestQuoteQuantums.String(),
-			),
-		)
+	// If `open_interest` >= `open_interest_upper_cap` where `upper_cap` is non-zero, OIMF is 1.
+	if oiCapUpper.Sign() > 0 && oiQuoteQuantums.Cmp(oiCapUpper) >= 0 {
+		return lib.BigU(lib.OneMillion)
 	}
 
-	// First, calculate base IMF in big.Rat
-	ratBaseIMF := new(big.Rat).SetFrac64(
-		int64(liquidityTier.InitialMarginPpm), // safe, since `InitialMargin` is uint32
-		int64(lib.OneMillion),
-	)
-
-	// `Effective IMF = Min(Base IMF + Max(IMF Increase, 0), 1.0)`
-	ratEffectiveIMF := ratBaseIMF.Add(
-		ratBaseIMF, // reuse pointer for memory efficiency
-		ratIMFIncrease,
-	)
-
-	// `Effective IMR = Effective IMF * Quote Quantums`
-	bigIMREffective := lib.BigRatRound(
-		ratEffectiveIMF.Mul(
-			ratEffectiveIMF, // reuse pointer for memory efficiency
-			ratQuoteQuantums,
-		),
-		true, // Round up initial margin.
-	)
+	// Get the ratio of where the current OI is between the lower and upper caps.
+	// The ratio should be between 0 and 1.
+	capNum := new(big.Int).Sub(oiQuoteQuantums, oiCapLower)
+	capDen := new(big.Int).Sub(oiCapUpper, oiCapLower)
 
-	// Return min(Effective IMR, Quote Quantums)
-	if bigIMREffective.Cmp(bigQuoteQuantums) >= 0 {
-		return bigQuoteQuantums
+	// Invariant checks.
+	if capNum.Sign() < 0 || capDen.Sign() <= 0 || capDen.Cmp(capNum) < 0 {
+		panic(fmt.Sprintf("invalid open interest values for liquidity tier %d", liquidityTier.Id))
 	}
-	return bigIMREffective
+	if capNum.Cmp(capDen) > 0 {
+		panic(errorsmod.Wrap(ErrOpenInterestLowerCapLargerThanUpperCap, lib.UintToString(liquidityTier.Id)))
+	}
+	if liquidityTier.InitialMarginPpm > lib.OneMillion {
+		panic(errorsmod.Wrap(ErrInitialMarginPpmExceedsMax, lib.UintToString(liquidityTier.Id)))
+	}
+
+	// Total IMF.
+	capScalePpm := lib.BigU(lib.OneMillion - liquidityTier.InitialMarginPpm)
+	result := new(big.Int).Mul(capScalePpm, capNum)
+	result.Div(result, capDen)
```

Comment:
This rounds towards zero right? Should we always round more conservatively (towards positive infinity) when calculating margin requirements? Think this is why `(OIMF) match collateralized at base IMF and just failed collateralization at OIMF` unit test is failing

Reply:
We should always round conservatively when doing the final calculation (imfPpm to quoteQuantums), but rounding conservatively in the case of increased ppm from OI doesn't seem entirely necessary to me.